core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -37,6 +37,10 @@
  */
 public final class UnsafeSorterSpillReader extends UnsafeSorterIterator implements Closeable {
   public static final int MAX_BUFFER_SIZE_BYTES = 16777216; // 16 mb
+  public static final int DEFAULT_BUFFER_SIZE_RATIO = 1024;
```

Comment:
Could you inline this default value in the config side (`createWithDefault(1024)`)?